### PR TITLE
Make project poetry based instead of setuptools

### PR DESCRIPTION
**Pull request by oharan2**
- Updated project metadata and dependencies
- Updated .release-it file `change-log` section
- Updated `poetry.lock` according new changes
- Update README

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -16,50 +16,46 @@ exclude = '''
...
+openshift = "^0.13.2"
+colorlog = "^6.7.0"
+pytest = "^7.4.0"
+pytest-testconfig = "^0.2.0"
+pyyaml = "^6.0.1"
+openshift-python-wrapper-data-collector = "^1.0.4"
+python-simple-logger = "^1.0.5"
+
+[tool.poetry.group.dev.dependencies]
+ipdb = "^0.13.13"
```
Comment (myakove) [inline pyproject.toml:57]: Please add `ipython = "*"` (this cannot be add via poetry add command)

Comment (oharan2) [reply pyproject.toml:57]: Done

**Review by myakove — CHANGES_REQUESTED** [review]
Please update `.release-it.json` file changelog to: 

`"changelog": "git log --pretty=format:\"* %s (%h) by %cn on %as\" ${from}...${to}"`

**Review by rnetser — APPROVED** [review]

**Review by rnetser — CHANGES_REQUESTED** [review]
requirements.txt should be removed

**Review by myakove — APPROVED** [review]

**Pull request merged**